image/tarexport/save.go

```diff
@@ -176,15 +176,12 @@ func (s *saveSession) save(outStream io.Writer) error {
 		reposFile := filepath.Join(tempDir, legacyRepositoriesFileName)
 		f, err := os.OpenFile(reposFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
 		if err != nil {
-			f.Close()
```

Comment:
Yes. This should be a `defer` after this if block. This `close` is wrong, but the `Close` won't be hit unless you're lucky enough to not hit any of the errors.